### PR TITLE
Sling 10644 return servlet resolver information in json format

### PR DESCRIPTION
**Pull request by angulito**
This PR adds support to return servlet resolver output in JSON format if the path request to the servlet resolver endpoint ends with `.json`. Output example for the URL parameter value `/example.myselector.ext`:
```
{
  "candidates": {
    "allowedServlets": [
      "org.apache.sling.pkg.impl.servlets.MyServlet"
    ],
    "deniedServlets": []
  },
  "method": GET,
  "warningMsg": "Note that in a real Sling request, the path might vary depending on the existence of resources that partially match it. This utility does not take this into account and uses the first dot to split between path and selectors/extension. As a workaround, you can replace dots with underline characters, for example, when testing such a URL.",
  "decomposedURL": {
    "path": "/example",
    "extension": "ext",
    "selectors": [
      "myselector"
    ]
  }
}
```

### CODE REVIEW

**Comment by cziegeler** [comment]
Thanks @angulito - the PR looks good to me, I'll leave it open to give other reviewers a chance

File: src/main/java/org/apache/sling/servlets/resolver/internal/console/WebConsolePlugin.java [hunk 1]
```diff
@@ -57,6 +43,28 @@
...
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * If the servlet request path ends with .json, the information is returned in JSON format.
+ * Otherwise, an HTML code is returned.
```
Comment (bdelacretaz) [inline src/main/java/org/apache/sling/servlets/resolver/internal/console/WebConsolePlugin.java:66]: Nitpick, I would say "an HTML document"

**Comment by bdelacretaz** [comment]
> ...Compose json by hand instead of adding a new dependency to manage it...

I think we need to define a nicer standard way of generating JSON in Sling, as we're doing that in many places.

In this case I agree that not requiring a new dependency is good - no need to change this code, with the clean jsonpath tests I think it's appropriate. But I feel bad about us requiring people to use `printf` to generate JSON...

**Comment by enapps-enorman** [comment]
> I think we need to define a nicer standard way of generating JSON in Sling, as we're doing that in many places.

The JsonGenerator from JSR 374 works fine and could be a good fit for any place generating or streaming JSON.

**Comment by bdelacretaz** [comment]
> The JsonGenerator from JSR 374 works fine...

Makes sense, thank you!

I have created a small sample at https://github.com/apache/sling-samples/tree/master/json-generation including generation with `JsonGenerator` and testing with `json-path`, I think that can be a useful "standard" for whenever we need to generate JSON.

This being said, I think this PR can stay as is using `printf` to generate JSON, as this is a core module it's good to minimize dependencies and given testing with `json-path` it's reasonable.

**Pull request merged**